doc/design/fully_static_graph.md

```diff
@@ -0,0 +1,122 @@
+# Design Doc: Fully Static Graph
+
+## Abstract
+
+We propose the *fully static graph* rule: training and inference must
+be fully specified by the static graph. This means training and
+inference should be able to run solely on the cpp core (no Python
+involved), everything should be implemented as an OP.
+
+The user can still use Python to achieve the same result for
+convenience when experimenting locally, but the distributed training
+will not support Python.
+
+## Background
+
+There are two paradigms for expressing the computation graph: dynamic
+and static. The dynamic paradigm constructs the graph on the fly:
+every time `eval` is called, a new graph is created. The static
```

Comment:
I didn't know what is `eval` until I read the Python code snippet few paragraphs below.

Reply:
Thanks! I have modified the design doc, this confusion no longer exists.